### PR TITLE
(maint) Prefer case exp to `send` in Acceptor

### PR DESCRIPTION
**Pull request by justinstoller**

### CODE REVIEW

File: lib/puppet/pops/validation.rb [hunk 1]
```diff
@@ -445,22 +445,14 @@ def prune(&block)
...
-      @warning_count += 1
-    end
-
-    def deprecation diagnostic
-      warning diagnostic
+      case diagnostic.severity
+      when :error
+        @error_count += 1
+      when :deprecation, :warning
+        @warning_count += 1
```
Comment (joshcooper) [inline lib/puppet/pops/validation.rb]: The `:deprecation` case used to call `warning` and didn't increment the `warning_count

Comment (justinstoller) [reply lib/puppet/pops/validation.rb]: Right. And the call to `warning` in `deprecation` would increment the `@warning_count`, right?

**Comment by justinstoller** [comment]
FWIW, this isn't super important. I was trying to understand how validation works in Puppet as part of evaluating the usages of `:strict` & `:strict_variables` for threadsafety of settings and have/will open up tickets that seem like easy wins.

I think the heavy usage of dynamic dispatch is confusing and possibly (though trivially) slower than simpler constructs. Let me know if I should just leave well enough alone.

File: lib/puppet/pops/validation.rb (hunk 1 above)
Comment (joshcooper) [reply lib/puppet/pops/validation.rb]: Oh that's confusing. I thought the previous call to `warning deprecation` was calling the puppet log function named `warning`. But what you have makes sense.

**Pull request merged**